### PR TITLE
Refactor appdata path logic

### PR DESCRIPTION
**Pull request by yufeih**
Consolidates all appdata path calculation logic to Appdata.cs,

_fixes_:
- git repository path should escape url
- file download path should respect query string

_renames_:
- RestoreUrl -> RestoreFile
- GetUrlRestorePath -> GetFileRestorePath

### CODE REVIEW

File: src/docfx/config/AppData.cs [hunk 1]
```diff
@@ -10,16 +10,48 @@ internal static class AppData
...
 
         public static string MutexDir => Path.Combine(AppDataDir, "mutex");
 
         public static string CacheDir => Path.Combine(AppDataDir, "cache");
 
         public static string GlobalConfigPath => GetGlobalConfigPath();
 
+        public static string GitHubUserCachePath => Path.Combine(CacheDir, "github-users.json");
+
+        public static string GetGitDir(string url)
```
Comment (OsmondJiang) [inline src/docfx/config/AppData.cs]: >GetGitDir [](start = 29, length = 9)

GetGitDir and GitDir may cause confusion

File: src/docfx/config/AppData.cs [hunk 2]
```diff
@@ -10,16 +10,48 @@ internal static class AppData
     {
         public static readonly string AppDataDir = GetAppDataDir();
 
-        public static string GitRestoreDir => Path.Combine(AppDataDir, "git");
+        public static string GitDir => Path.Combine(AppDataDir, "git");
```
Comment (OsmondJiang) [inline src/docfx/config/AppData.cs]: GitRootDir to avoid confusion?

File: src/docfx/config/AppData.cs [hunk 3]
```diff
@@ -8,33 +8,65 @@ namespace Microsoft.Docs.Build
...
+
+        public static string GetFileDownloadDir(string url)
+        {
+            // URL to a resource is case sensitive, query string matters, so hash the download path
+            return PathUtility.NormalizeFolder(Path.Combine(DownloadsRoot, UrlToPath(url) + "-" + url.Trim().GetMd5HashShort()));
+        }
+
+        public static string GetCommitCachePath(string remote)
+        {
+            return Path.Combine(AppData.CacheRoot, "commits", HashUtility.GetMd5Hash(remote));
```
Comment (live1206) [inline src/docfx/config/AppData.cs]: >AppData [](start = 32, length = 7)

can be removed

Comment (yufeih) [reply src/docfx/config/AppData.cs]: yes

**Pull request merged**